### PR TITLE
Remove Duplicate Code in SusiServer.java

### LINKED ISSUE #560: Removed Duplicate Code
In the file `SusiServer.java` there is a duplicate exists for checking availability of Port.

### PR DESCRIPTION
**Pull request by yashLadha**
Fixes #560

### CODE REVIEW

File: src/ai/susi/SusiServer.java [hunk 1]
```diff
@@ -626,33 +626,27 @@ private static void checkServerPorts(int httpPort, int httpsPort) throws IOExcep
...
-	            sss.setReuseAddress(true);
-	            sss.setReceiveBufferSize(65536);
-	        } catch (IOException e) {
-	            // the socket is already occupied by another service
-	        	throw new IOException("port " + httpsPort + " is already occupied by another service, maybe another SUSI is running on this port already. exit.");
-	        } finally {
-	            // close the socket again
-	            if (sss != null) sss.close();
-	        }
+            checkPort(httpsPort, sss);
```
Comment (yukiisbored) [inline src/ai/susi/SusiServer.java]: We should do this only if https is enabled.

File: src/ai/susi/SusiServer.java [hunk 2]
```diff
@@ -626,33 +626,27 @@ private static void checkServerPorts(int httpPort, int httpsPort) throws IOExcep
...
-	            ss.setReuseAddress(true);
-	            ss.setReceiveBufferSize(65536);
-	        } catch (IOException e) {
-	            // the socket is already occupied by another service
-	            throw new IOException("port " + httpPort + " is already occupied by another service, maybe another SUSI is running on this port already. exit.");
-	        } finally {
-	            // close the socket again
-	            if (ss != null) ss.close();
-	        }
+            checkPort(httpPort, ss);
```
Comment (yukiisbored) [inline src/ai/susi/SusiServer.java]: We should do this when it's not on https only mode.

Comment (yukiisbored) [inline src/ai/susi/SusiServer.java]: Sorry for nitpicking, fix the indentation please.

File: src/ai/susi/SusiServer.java (hunk 1 above)
Comment (yukiisbored) [inline src/ai/susi/SusiServer.java]: Same as https://github.com/fossasia/susi_server/pull/561/files?diff=split#diff-d93e29e1c875d30ce3d763c19cce4084R629

**Comment by yashLadha** [comment]
I will make those changes

File: src/ai/susi/SusiServer.java (hunk 1 above)
Comment (yashLadha) [reply src/ai/susi/SusiServer.java]: Can you explain more? @yukiisbored

**Comment by yukiisbored** [comment]
@yashLadha Please squash your commits.

**Comment by yukiisbored** [comment]
On https://github.com/fossasia/susi_server/pull/561/commits/17a58095ce9ea877478308aa6671680277e59382, We don't need all of them to be fixed. That could be outside this PR's scope. Just align the identations with the old ones. We can care about code formatting later.

**Comment by yashLadha** [comment]
@yukiisbored is it fine now?

**Comment by yukiisbored** [comment]
Please remove the messy "merge commits". Please use `rebase` not `merge` when there's updates on another branch. 

Right now, I recommend doing this:
1. Create a new branch with this current branch as the base, this will be your old branch as a "backup"
2. Destroy the current branch
3. Create a new branch with *the same name* as the one stated in this PR with latest master as the base
4. Cherry-pick your commits from the old "backup branch"
5. Do a force push

**Comment by yashLadha** [comment]
@yukiisbored Done

**Comment by mariobehling** [comment]
Any idea why Travis is not running? https://travis-ci.org/fossasia/susi_server/builds/276836225?utm_source=github_status&utm_medium=notification

**Comment by DravitLochan** [comment]
@mariobehling It ran and passed. I think it is good to go.

**Comment by hpdang** [comment]
Thank you!

**Pull request merged**